airflow/contrib/hooks/sagemaker_hook.py

```diff
@@ -0,0 +1,811 @@
+        """
+
+        return self.conn\
+            .describe_hyper_parameter_tuning_job(
+                HyperParameterTuningJobName=name)
+
+    def describe_model(self, name):
+        """
+        :param name: the name of the SageMaker model
+        :type name: string
+        Return the SageMaker model info associated with the model name
+        :return: A dict contains all the model info
+        """
+
+        return self.conn\
+            .describe_model(
+                ModelName=name)
+
+    def describe_transform_job(self, name):
+        """
+        :param name: the name of the transform job
+        :type name: string
+        Return the transform job info associated with the current job_name
+        :return: A dict contains all the transform job info
+        """
+
+        return self.conn\
+            .describe_transform_job(
+                TransformJobName=name)
+
+    def describe_endpoint_config(self, name):
+        """
+        :param name: the name of the endpoint config
+        :type name: string
+        :return: A dict contains all the endpoint config info
+        """
+
+        return self.conn\
+            .describe_endpoint_config(
+                EndpointConfigName=name)
+
+    def describe_endpoint(self, name):
+        """
+        :param name: the name of the endpoint
+        :type name: string
+        :return: A dict contains all the endpoint info
+        """
+
+        return self.conn\
+            .describe_endpoint(
+                EndpointName=name)
+
+    def check_status(self, job_name, non_terminal_states,
+                     failed_states, key,
+                     describe_function, check_interval,
+                     max_ingestion_time):
+        """
+        Check status of a SageMaker job
+        :param job_name: name of the job to check status
+        :type job_name: str
+        :param non_terminal_states: the set of non_terminal states
+        :type non_terminal_states: set
+        :param failed_states: the set of failed states
+        :type failed_states: set
+        :param key: the key of the response dict
+        that points to the state
+        :type key: str
+        :param describe_function: the function used to retrieve the status
+        :type describe_function: python callable
+        :param args: the arguments for the function
+        :param check_interval: the time interval in seconds which the operator
+        will check the status of any SageMaker job
+        :type check_interval: int
+        :param max_ingestion_time: the maximum ingestion time in seconds. Any
+        SageMaker jobs that run longer than this will fail. Setting this to
+        None implies no timeout for any SageMaker job.
+        :type max_ingestion_time: int
+        :return: response of describe call after job is done
+        """
+        sec = 0
+        running = True
+
+        while running:
+            time.sleep(check_interval)
```

Comment:
@ashb Note to self: Come back to this once you've reviewed the rest. We might not want the loop in here, but to have it in a sensor instead.